src/PrestaShopBundle/Install/XmlLoader.php

```diff
@@ -177,10 +177,9 @@ public function setIds($ids)
     public function getSortedEntities()
     {
         // Browse all XML files from data/xml directory
-        $entities = array();
-        $dependencies = array();
-        $fd = opendir($this->data_path);
-        while ($file = readdir($fd)) {
+        $entities = [];
+        $dependencies = [];
+        foreach (scandir($this->data_path) as $file) {
```

Comment:
Hi @PierreRambaud 
Is it possible to use Tools Classe here?
```php
Tools::scandir($this->data_path,'xml')
```
See https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Tools.php#L3768 

Can you allow me to refactor this class please ? Use `Tools` instead of `scandir` + `preg_match`?

Reply:
IMHO, it's not a good idea, this class must be outside the Core because it's only related to the install process. And the `Tools::scandir` doesn't provide great value :/

Reply:
Ok Thanks you @PierreRambaud